### PR TITLE
[chore] add promtail receiver to otelcontribcol

### LINKED ISSUE #17082: receiver/promtailreceiver not in release v0.67.0
### Component(s)

receiver/promtail

### Describe the issue you're reporting

Promtail-receiver is nowhere to be found in release v0.67.0. (https://github.com/open-telemetry/opentelemetry-collector-releases/blob/v0.67.0/distributions/otelcol-contrib/manifest.yaml)

But the release-description says it is. (https://github.com/open-telemetry/opentelemetry-collector-contrib/releases/tag/v0.67.0)

Would be nice if you could include it as soon as possible :)

### PR DESCRIPTION
**Pull request by mar4uk**
**Description:** <Describe what has changed.>
Add promtail receiver to otelcontribcol

**Link to tracking Issue:** fix https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/17082

### CODE REVIEW

**Review by jpkrohling — APPROVED** [review]

File: receiver/promtailreceiver/config.go [hunk 1]
```diff
@@ -52,9 +52,9 @@ func NewConfigWithID(operatorID string) *Config {
 }
 
 type PromtailInputConfig struct {
-	PositionsConfig positions.Config      `mapstructure:"positions,omitempty" yaml:"positions,omitempty"`
-	ScrapeConfig    []scrapeconfig.Config `mapstructure:"scrape_configs,omitempty" yaml:"scrape_configs,omitempty"`
-	TargetConfig    file.Config           `mapstructure:"target_config,omitempty" yaml:"target_config,omitempty"`
+	PositionsConfig positions.Config      `mapstructure:"-" yaml:"positions,omitempty"`
+	ScrapeConfig    []scrapeconfig.Config `mapstructure:"-" yaml:"scrape_configs,omitempty"`
+	TargetConfig    file.Config           `mapstructure:"-" yaml:"target_config,omitempty"`
 }
```
Comment (bogdandrutu) [inline receiver/promtailreceiver/config.go:58]: Why this change?

Comment (mar4uk) [reply receiver/promtailreceiver/config.go:58]: The test https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/54da8bd95661636d0639b1b2bd2832a391d406ee/cmd/otelcontribcol/components_test.go#L12 failed because inner structs don't have `mapstructure` tags, they only have `yaml` tags.

Since promtail config has its own Unmarshall method that uses `yaml.Marshall`, `yaml.Unmarshall` methods, `mapstructure` tag is not needed here
